synapse/events/utils.py

```
@@ -164,6 +256,10 @@ def serialize_event(e, time_now_ms, as_client_event=True,
                 d["unsigned"]["transaction_id"] = txn_id
 
     if as_client_event:
-        return event_format(d)
-    else:
-        return d
+        d = event_format(d)
+
+    if (only_event_fields and isinstance(only_event_fields, list) and
+            all(isinstance(f, basestring) for f in only_event_fields)):
```

If the args are incorrect this should raise an exception.